web/services/nft_service.py

```
@@ -0,0 +1,243 @@
+                        return f"ipfs://{ipfs_hash}"
+                    else:
+                        logger.warning(f"Pinning failed: {pin_response.text}")
+                        return f"ipfs://{ipfs_hash}"
+
+                else:
+                    # Log detailed error
+                    logger.error(f"Upload failed: {response.status_code} - {response.text}")
+                    raise ValueError(f"Upload failed: {response.text}")
+
+        except Exception as e:
+            logger.error(f"IPFS Upload Error: {e}")
```

Ruff (0.8.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)